compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1581,6 +1581,10 @@ private boolean validateErrorVariable(BLangErrorVariable errorVariable) {
         errorVariable.type = errorType;
 
         if (!errorVariable.isInMatchStmt) {
+            if (errorVariable.message == null) {
+                dlog.error(errorVariable.pos, DiagnosticErrorCode.INVALID_ERROR_BINDING_PATTERN, errorVariable.type);
+                return false;
+            }
```

Comment:
Spec allows error binding patterns without the message

Reply:
Fixed in 8aaa9bc